File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/utils/Injection.java
package com.non_name_hero.calenderview.utils;

import android.content.Context;

import androidx.annotation.NonNull;

import com.non_name_hero.calenderview.data.source.ScheduleRepository;
import com.non_name_hero.calenderview.data.source.local.PigLeadDatabase;
import com.non_name_hero.calenderview.data.source.local.ScheduleDataLocalSource;
import com.non_name_hero.calenderview.data.source.remote.ScheduleDataRemoteSource;

import static com.google.android.gms.common.internal.Preconditions.checkNotNull;

public class Injection {
    public static ScheduleRepository provideScheduleRepository(@NonNull Context context) {
        checkNotNull(context);
        PigLeadDatabase pigLeadDatabase = PigLeadDatabase.getInstance(context);
        AppExecutors appExecutors = new AppExecutors();

        ScheduleRepository scheduleRepository = ScheduleRepository.getInstance(
                ScheduleDataLocalSource.getInstance(
                        appExecutors,
                        pigLeadDatabase.scheduleDao()),
                ScheduleDataRemoteSource.getInstance(appExecutors));
        return scheduleRepository;
    }
}

File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/utils/Injection.kt
package com.non_name_hero.calenderview.utils

import android.content.Context
import com.google.android.gms.common.internal.Preconditions
import com.non_name_hero.calenderview.data.source.ScheduleRepository
import com.non_name_hero.calenderview.data.source.local.PigLeadDatabase.Companion.getInstance
import com.non_name_hero.calenderview.data.source.local.ScheduleDataLocalSource
import com.non_name_hero.calenderview.data.source.remote.ScheduleDataRemoteSource

object Injection {
    fun provideScheduleRepository(context: Context): ScheduleRepository {
        Preconditions.checkNotNull(context)
        val pigLeadDatabase = getInstance(context)
        val appExecutors = AppExecutors()
        return ScheduleRepository.getInstance(
                ScheduleDataRemoteSource(),
                ScheduleDataLocalSource(
                        AppExecutors(),
                        pigLeadDatabase.scheduleDao()))
    }
}